### DISCUSSION TITLE
Using my own custom persistentVolumeClaim

### DISCUSSION THREAD

**Discussion by miketoupin** [secureCodeBox/secureCodeBox#1659] [General]
Good day,

Rather than using the default Minio storage solution or S3, I would like to use the preferred way to do this in K8S, which is via CSI driver.  I can create my own StorageClass and PersistenVolume, all I need is a way to tell the Operator to use the PVC for the storage.  How can I do this to replace the S3/Minio setup? Thx!

**Comment by J12934** [comment]
Hi @miketoupin 

That's not really possible as we use minio / s3 for more then "just" the storage but also as a way to give a very limited amount of access to individual pods using presigned urls. (e.g. for the scan pod to only be able to persist the results of the scan into storage)

This isn't really doable with PVC as we'd have to create a new PVC per scan to get the same amount of control from it.

So basically if you want to store them in a PVC we'd recommend using a minio setup. 
SCB will connect to minio, minio then stores it in the PVC.
